core/src/main/scala/org/http4s/headers/Allow.scala

```diff
@@ -17,14 +17,20 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import org.http4s.internal.parsing.Rfc7230
 import org.http4s.util.Writer
 
 object Allow extends HeaderKey.Internal[Allow] with HeaderKey.Singleton {
   def apply(ms: Method*): Allow = Allow(ms.toSet)
 
   override def parse(s: String): ParseResult[Allow] =
-    HttpHeaderParser.ALLOW(s)
+    ParseResult.fromParser(parser, "Invalid Allow")(s)
+  private[http4s] val parser = Rfc7230
```

Comment:
alternative version: `  private[http4s] val parser = Method.parser
    .repSep0(Parser.char(',').surroundedBy(Rfc7230.ows))
    .map(list => Allow(list.toSet))`

Reply:
hmm not sure if this actually equivivalent

https://tools.ietf.org/html/rfc7230#section-7

allows e.g. `"foo , ,bar,charlie   "` as a valid value for the `#element` syntax

the header is just weirdly specified https://tools.ietf.org/html/rfc7231#section-7.4.1
the syntax is described as `Allow = #method` which would mean at least one method, but they immediately contradict the syntax: `An empty Allow field value indicates that the resource allows no methods`

Reply:
hm, interesting. maybe we need a test that also adds that edge-case.